programs/bpf_loader/src/syscalls.rs

```diff
@@ -76,6 +78,40 @@ pub enum SyscallError {
     TooManyAccounts,
     #[error("Overlapping copy")]
     CopyOverlapping,
+    #[error("BigNumber: Modular exponentiation error {0}")]
+    BigNumberModExpError(String),
+    #[error("BigNumber: Bytes would exceed buffer provided")]
+    BigNumberToBytesError,
+    #[error("BigNumber: expects {0} args but provided {1}")]
+    BigNumberArgError(u64, u64),
+    #[error("BigNumber: add error {0}")]
+    BigNumberAddError(String),
+    #[error("BigNumber: sub error {0}")]
+    BigNumberSubError(String),
+    #[error("BigNumber: mul error {0}")]
+    BigNumberMulError(String),
+    #[error("BigNumber: div error {0}")]
+    BigNumberDivError(String),
+    #[error("BigNumber: exp error {0}")]
+    BigNumberExpError(String),
+    #[error("BigNumber: sqr error {0}")]
+    BigNumberSqrError(String),
+    #[error("BigNumber: mod_sqr error {0}")]
+    BigNumberModSqrError(String),
+    #[error("BigNumber: mod_mul error {0}")]
+    BigNumberModMulError(String),
+    #[error("BigNumber: mod_inv error {0}")]
+    BigNumberModInvError(String),
+    #[error("BigNumber: error in calling OpenSSL BigNum {0}")]
+    BigNumberOpenSSLError(String),
```

Comment:
I'm not sure we should have openssl specific errors. We want to be able to swap out openssl for another bignum implementation, should the need arise.

Reply:
Yeah, renaming and redescribing them to be generic should be fine, just can't report the actual openssl error if known :-(

Reply:
Actually, the string in `BigNumberOpenSSLError(String)` is the OpenSSL reason if they provide it. This would be a matter of just changing the main error ID into something more generic like `BigNumberMathOperationError(String)`

And the the encoding concerns, just tests in "C"? or both "C" and "Rust"?

Reply:
Wouldn't just a test in `syscalls.rs` satisfy the encoding question? i.e.:
```rust
let le_byte_array = u32::to_le_bytes(256); // [0,0,1,255]
assert_neq!(OpenSSL::from_slice(&le_byte_array).unwrap.to_vec(), [1, 255]);
```

Reply:
I've already tested that leading zeros into OpenSSL strips the leading zeros but I can add a test as well

Reply:
Yeah, the tests could be unit tests in `syscalls.rs`